sdk/notificationhubs/notification-hubs/src/utils/connectionStringUtils.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createSasTokenProvider, parseConnectionString, SasTokenProvider } from "@azure/core-amqp";
+
+/**
+ * The set of properties that comprise a Notification Hubs connection string.
+ */
+export interface NotificationHubsConnectionStringProperties {
```

Comment:
the typing doesn't quite feel accurate, since really it's `{ endpoint: string } | { endpoint: string; sharedAccessKey: string; sharedAccessKeyName: string }`

Reply:
Made the others non-optional as all three are required.